File: lab2zad1.java
public class Samochod {
    private String marka;
    private String model;
    private int rokProdukcji;

    public Samochod(String marka, String model, int rokProdukcji) {
        this.marka = marka;
        this.model = model;
        this.rokProdukcji = rokProdukcji;
    }

    public Samochod() {
        this.marka = "Nieznana";
        this.model = "Nieznany";
        this.rokProdukcji = 0;
    }

    public void displayInfo() {
        System.out.println("Marka: " + marka + ", Model: " + model + ", Rok produkcji: " + rokProdukcji);
    }

    public static void main(String[] args) {
        Samochod samochod1 = new Samochod("Toyota", "Corolla", 2020);
        samochod1.displayInfo();

        Samochod samochod2 = new Samochod();
        samochod2.displayInfo();
    }
}

File: lab2zad1.cpp
#include <iostream>
#include <string>
using namespace std;

class Samochod {
private:
    string marka;
    string model;
    int rokProdukcji;

public:
    Samochod(string marka, string model, int rokProdukcji)
        : marka(marka), model(model), rokProdukcji(rokProdukcji) {}

    Samochod()
        : marka("Nieznana"), model("Nieznany"), rokProdukcji(0) {}

    ~Samochod() {
        cout << "Destruktor wywołany dla: " << marka << " " << model << endl;// informacja o likwidacji obiektu?
    }

    void displayInfo() const {
        cout << "Marka: " << marka << ", Model: " << model << ", Rok produkcji: " << rokProdukcji << endl;
    }
};

int main() {
    Samochod samochod1("Toyota", "Corolla", 2020);
    samochod1.displayInfo();

    Samochod samochod2;
    samochod2.displayInfo();

    return 0;
}
